metadata-ingestion/src/datahub/ingestion/source/powerbi/powerbi.py

```diff
@@ -853,24 +867,36 @@ class PowerBiDashboardSource(Source):
     source_config: PowerBiDashboardSourceConfig
     reporter: PowerBiDashboardSourceReport
     accessed_dashboards: int = 0
+    platform: str = "powerbi"
 
     def __init__(self, config: PowerBiDashboardSourceConfig, ctx: PipelineContext):
-        super().__init__(ctx)
+        super(PowerBiDashboardSource, self).__init__(config, ctx)
         self.source_config = config
         self.reporter = PowerBiDashboardSourceReport()
-        self.auth_token = PowerBiAPI(self.source_config).get_access_token()
         self.powerbi_client = PowerBiAPI(self.source_config)
         self.mapper = Mapper(config, self.reporter)
 
+        # Create and register the stateful ingestion use-case handler.
+        self.stale_entity_removal_handler = StaleEntityRemovalHandler(
+            source=self,
+            config=self.source_config,
+            state_type_class=BaseSQLAlchemyCheckpointState,
+            pipeline_name=ctx.pipeline_name,
+            run_id=ctx.run_id,
+        )
+
+    def get_platform_instance_id(self) -> str:
+        return self.source_config.platform_name
```

Comment:
Where is this used ? Typically platform_instance_id refers to `platform_instance` field of config.

Reply:
It is used by stateful classes